spec/poisson_spec.rb

```diff
@@ -1,6 +1,36 @@
 require File.expand_path(File.dirname(__FILE__) + '/spec_helper.rb')
 include ExampleWithGSL
 describe Distribution::Poisson do
+  shared_examples_for 'poisson engine(with rng)' do
+    it 'should return correct rng' do
+      lambda_val = 2
+      exp_mean = 2
```

Comment:
I don't have a strong opinion about using a fixed value to test this, but I'd randomize the mean to be picked from a specific range, and derive the other values from it given there are simple formulae for doing so.

Doing so would enable you to be more confident about the property being of the distribution being held since you'd be testing an invariant.

Reply:
Thank you for helpful comments.
I had changed to used `rand ()` function for getting more confident that.